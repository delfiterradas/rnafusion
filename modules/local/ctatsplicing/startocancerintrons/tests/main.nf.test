nextflow_process {

    name "Test Process CTATSPLICING_STARTOCANCERINTRONS"
    script "../main.nf"
    process "CTATSPLICING_STARTOCANCERINTRONS"
    options "-stub"

    test("test without BAM") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [id:"test"],
                    file("test.SJ.out.tab"),
                    file("test.Chimeric.out.junctions"),
                    [],
                    []
                ]
                input[1] = [
                    [id:"reference"],
                    file("ctat_genome_lib")
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.findAll { key, value -> !key.isNumber() }).match() }
            )
        }
    }

    test("test with BAM") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [id:"test"],
                    file("test.SJ.out.tab"),
                    file("test.Chimeric.out.junctions"),
                    file("test.Aligned.sortedByCoord.out.bam"),
                    []
                ]
                input[1] = [
                    [id:"reference"],
                    file("ctat_genome_lib")
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.findAll { key, value -> !key.isNumber() }).match() }
            )
        }
    }
}
