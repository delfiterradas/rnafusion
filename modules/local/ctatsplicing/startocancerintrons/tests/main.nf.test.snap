{
    "test without BAM": {
        "content": [
            {
                "cancer_introns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer.introns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_prelim": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer.introns.prelim:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_sorted_bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer_intron_reads.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_sorted_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer_intron_reads.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "chckpts": [
                    [
                        {
                            "id": "test"
                        },
                        "test.chckpts:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_reads_sorted_bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_reads.sorted.sifted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_reads_sorted_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_reads.sorted.sifted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "igv_html": [
                    
                ],
                "igv_tracks": [
                    
                ],
                "introns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.introns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "introns_igv_bed": [
                    
                ],
                "versions": [
                    "versions.yml:md5,fcf861a15f9951342a874b6bc476a37e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-17T13:35:13.723215847"
    },
    "test with BAM": {
        "content": [
            {
                "cancer_introns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer.introns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_prelim": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer.introns.prelim:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_sorted_bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer_intron_reads.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cancer_introns_sorted_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cancer_intron_reads.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "chckpts": [
                    [
                        {
                            "id": "test"
                        },
                        "test.chckpts:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_reads_sorted_bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_reads.sorted.sifted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_reads_sorted_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_reads.sorted.sifted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "igv_html": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ctat-splicing.igv.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "igv_tracks": [
                    [
                        {
                            "id": "test"
                        },
                        "test.igv.tracks:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "introns": [
                    [
                        {
                            "id": "test"
                        },
                        "test.introns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "introns_igv_bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.introns.for_IGV.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fcf861a15f9951342a874b6bc476a37e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-17T13:33:27.36677449"
    }
}